src/snap_name_of.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2021 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * under the terms of the GNU General Public License version 2 or 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: William Wild <william.wold@canonical.com>
+ */
+
+#include "snap_name_of.h"
+
+#include <unistd.h>
+#include <string.h>
+#include <iostream>
+#include <miral/version.h>
+#include <sys/apparmor.h>
+
+auto snap_name_of(miral::Application const& app) -> std::string
+{
+#if MIRAL_VERSION >= MIR_VERSION_NUMBER(3, 4, 0)
+    int const app_fd = miral::socket_fd_of(app);
+#else
+    int const app_fd = -1;
+#endif
+    char* label_cstr;
+    char* mode_cstr;
+    errno = 0;
+    if (app_fd < 0)
+    {
+        return "";
+    }
+    else if (aa_getpeercon(app_fd, &label_cstr, &mode_cstr) < 0)
+    {
+        std::cerr
+            << "aa_getpeercon() failed for process " << miral::pid_of(app)
+            << ": " << strerror(errno)
+            << std::endl;
```

Comment:
```suggestion
        mir::log_debug("aa_getpeercon() failed for process %d : %s", miral::pid_of(app), strerror(errno));
```

(This also requires `MIR_LOG_COMPONENT`, which should probably go in `CMakeLists.txt`:
```cmake
target_compile_definitions(frame PRIVATE MIR_LOG_COMPONENT="frame")
```

Reply:
Done